src/rdkafka_telemetry_encode.c

```diff
@@ -0,0 +1,837 @@
+        if (!pb_encode_tag_for_field(stream, field))
+                return false;
+
+        return pb_encode_submessage(
+            stream, opentelemetry_proto_metrics_v1_ResourceMetrics_fields,
+            resource_metrics);
+}
+
+static bool encode_key_value(pb_ostream_t *stream,
+                             const pb_field_t *field,
+                             void *const *arg) {
+        if (!pb_encode_tag_for_field(stream, field))
+                return false;
+        opentelemetry_proto_common_v1_KeyValue *key_value =
+            (opentelemetry_proto_common_v1_KeyValue *)*arg;
+        return pb_encode_submessage(
+            stream, opentelemetry_proto_common_v1_KeyValue_fields, key_value);
+}
+
+static bool encode_key_values(pb_ostream_t *stream,
+                              const pb_field_t *field,
+                              void *const *arg) {
+        rd_kafka_telemetry_key_values_repeated_t *kv_arr =
+            (rd_kafka_telemetry_key_values_repeated_t *)*arg;
+        size_t i;
+
+        for (i = 0; i < kv_arr->count; i++) {
+
+                opentelemetry_proto_common_v1_KeyValue *kv =
+                    kv_arr->key_values[i];
+                if (!pb_encode_tag_for_field(stream, field))
+                        return false;
+
+                if (!pb_encode_submessage(
+                        stream, opentelemetry_proto_common_v1_KeyValue_fields,
+                        kv))
+                        return false;
+        }
+        return true;
+}
+
+static void free_metrics(
+    opentelemetry_proto_metrics_v1_Metric **metrics,
+    char **metric_names,
+    opentelemetry_proto_metrics_v1_NumberDataPoint **data_points,
+    opentelemetry_proto_common_v1_KeyValue *datapoint_attributes_key_values,
+    size_t count) {
+        size_t i;
+        for (i = 0; i < count; i++) {
+                rd_free(data_points[i]);
+                rd_free(metric_names[i]);
+                rd_free(metrics[i]);
+        }
+        rd_free(data_points);
+        rd_free(metric_names);
+        rd_free(metrics);
+        rd_free(datapoint_attributes_key_values);
+}
+
+static void free_resource_attributes(
+    opentelemetry_proto_common_v1_KeyValue **resource_attributes_key_values,
+    rd_kafka_telemetry_resource_attribute_t *resource_attributes_struct,
+    size_t count) {
+        size_t i;
+        if (count == 0)
+                return;
+        for (i = 0; i < count; i++)
+                rd_free(resource_attributes_key_values[i]);
+        rd_free(resource_attributes_struct);
+        rd_free(resource_attributes_key_values);
+}
+
+static void serialize_Metric(
+    rd_kafka_t *rk,
+    rd_kafka_broker_t *rkb,
+    const rd_kafka_telemetry_metric_info_t *info,
+    opentelemetry_proto_metrics_v1_Metric **metric,
+    opentelemetry_proto_metrics_v1_NumberDataPoint **data_point,
+    opentelemetry_proto_common_v1_KeyValue *data_point_attribute,
+    rd_kafka_telemetry_metric_value_calculator_t metricValueCalculator,
```

Comment:
You can use the same case as the type
```suggestion
    rd_kafka_telemetry_metric_value_calculator_t metric_value_calculator,
```

Reply:
Updated